src/operator/nn/batch_norm.cc

```diff
@@ -164,25 +164,65 @@ void BatchNormForwardImpl(mshadow::Stream<cpu> *,
 
     // note that var is still invstd
     if (!param_.fix_gamma) {
-      if (IsBNWriting(req[batchnorm::kData])) {
-        ForEachFast(inputData, outputData, channel,
-                    [thisWeight, thisBias, thisMean, thisInvstd](const DType *in_data,
-                                                                 DType *out_data) {
-                      *out_data = static_cast<DType>(
-                        ((*in_data - thisMean) * thisInvstd) * thisWeight + thisBias);
-                    });
+      if (!param_.fix_beta) {
```

Comment:
Do you really need 4 cases rather than collate 2 cases separately?

Reply:
It is basically 2 cases based on fix_gamma. fix_beta cases are collated within it.
This has more readability with flags on the cases. Suggestions?

Reply:
Fix gamma and beta only involves whether including (*gamma) and (+bias), but here's more than 50 lines only to accomplish that. It should be fine with the current lambda function with one or two cases, but with 4 cases or maybe 8 cases in the future, it's going to cause a lot of trouble.